File: structures/queue/queueArray.py
class QueueArray:
    def __init__(self):
        self._capacity = 8
        self._fila = [None] * self._capacity
        self._count = 0
        self._primeiro = 0
        self._ultimo = 0

    def enqueue(self, element):
        if self._count == self._capacity:
            self._resize(2 * self._capacity)

        self._fila[self._ultimo] = element
        self._ultimo = (self._ultimo + 1) % self._capacity
        self._count += 1

    def dequeue(self):
        if self.is_empty():
            raise RuntimeError("A fila está vazia!")

        element = self._fila[self._primeiro]
        self._fila[self._primeiro] = None
        self._primeiro = (self._primeiro + 1) % self._capacity
        self._count -= 1
        return element

    def peek(self):
        if self.is_empty():
            raise RuntimeError("A fila está vazia!")
        return self._fila[self._primeiro]

    def is_empty(self):
        return self._count == 0

    def size(self):
        return self._count

    def clear(self):
        self.__init__()

    def _resize(self, new_capacity):
        nova_fila = [None] * new_capacity
        for i in range(self._count):
            nova_fila[i] = self._fila[(self._primeiro + i) % self._capacity]
        self._fila = nova_fila
        self._primeiro = 0
        self._ultimo = self._count
        self._capacity = new_capacity

File: structures/queue/queueArray.java
public class QueueArray<T> {
    private T[] queue;
    private int capacity;
    private int count;
    private int first;
    private int last;
    
    @SuppressWarnings("unchecked")
    public QueueArray() {
        this.capacity = 8;
        this.queue = (T[]) new Object[this.capacity];
        this.count = 0;
        this.first = 0;
        this.last = 0;
    }
    
    public void enqueue(T element) {
        if (this.count == this.capacity) {
            resize(2 * this.capacity);
        }
        this.queue[this.last] = element;
        this.last = (this.last + 1) % this.capacity;
        this.count++;
    }
    
    public T dequeue() {
        if (isEmpty()) {
            throw new RuntimeException("Queue is empty!");
        }
        T element = this.queue[this.first];
        this.queue[this.first] = null;
        this.first = (this.first + 1) % this.capacity;
        this.count--;
        return element;
    }
    
    public T peek() {
        if (isEmpty()) {
            throw new RuntimeException("Queue is empty!");
        }
        return this.queue[this.first];
    }
    
    public boolean isEmpty() {
        return this.count == 0;
    }
    
    public int size() {
        return this.count;
    }
    
    @SuppressWarnings("unchecked")
    public void clear() {
        this.capacity = 8;
        this.queue = (T[]) new Object[this.capacity];
        this.count = 0;
        this.first = 0;
        this.last = 0;
    }
    
    @SuppressWarnings("unchecked")
    private void resize(int newCapacity) {
        T[] newQueue = (T[]) new Object[newCapacity];
        for (int i = 0; i < this.count; i++) {
            newQueue[i] = this.queue[(this.first + i) % this.capacity];
        }
        this.queue = newQueue;
        this.first = 0;
        this.last = this.count;
        this.capacity = newCapacity;
    }
}